test/test_base_metric.py

```diff
@@ -4,71 +4,170 @@
 import metric_learn
 import numpy as np
 from sklearn import clone
-from sklearn.utils.testing import set_random_state
+import sklearn
+from packaging import version
 from test.test_utils import ids_metric_learners, metric_learners, remove_y
+skversion = version.parse(sklearn.__version__)
+if skversion >= version.parse('0.22.0'):
+    from sklearn.utils._testing import set_random_state
+else:
+    from sklearn.utils.testing import set_random_state
 
 
 def remove_spaces(s):
   return re.sub(r'\s+', '', s)
 
 
+def sk_repr_kwargs(def_kwargs, nndef_kwargs):
+    """Given the non-default arguments, and the default
+    keywords arguments, build the string that will appear
+    in the __repr__ of the estimator, depending on the
+    version of scikit-learn.
+    """
+    if skversion >= version.parse('0.22.0'):
+        def_kwargs = ""
+    nndef_kwargs = eval(f"dict({nndef_kwargs})")
```

Comment:
Rather than using eval, could we pass actual dicts here instead? The values are all simple literals, so it should be easier that way.

Reply:
Yes you're right, will do

Reply:
done